cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java

```diff
@@ -501,4 +501,25 @@ void setWorkflowProgramStart(Id.Program programId, String programRunId, String w
    */
   WorkflowStatistics getWorkflowStatistics(Id.Workflow workflowId, long startTime,
                                            long endTime, List<Double> percentiles);
+
+  /**
+   * Returns detailed statistics for a run of a workflow.
+   *
+   * @param workflowId The Workflow whose run needs to be queried
+   * @param runId RunId of the workflow run
+   * @return A detailed report of the statistics
+   */
+  WorkflowDataset.WorkflowRunRecord getWorkflowRun(Id.Workflow workflowId, String runId);
```

Comment:
I think you could pass in ``Id.Run`` here, but its ok to keep this as-is as well, since you want to enforce that it only supports workflows